### PR TITLE
Delete file volume func implementation

### PR DESCRIPTION
**Pull request by qingfeng0820**
For file volume, the returned id's format is filesystemId:NFSShareId. And in k8s the corresponding file volume id's format is cis-unity-file-filesystemId:NFSShare.

### CODE REVIEW

File: service/controller.go [hunk 1]
```diff
@@ -14,6 +19,65 @@ const (
...
+
+var (
+	// BlockVolumePrefix is a prefix of Block volume's ID.
+	BlockVolumePrefix = fmt.Sprintf("%s-block-", Name)
+	// FileVolumePrefix is a prefix of File volume's ID.
+	FileVolumePrefix = fmt.Sprintf("%s-file-", Name)
+)
+
+// getBackendIdAndTypeByVolumeId is used to get id and volume type by volume id
+// if it is block type, backend id is block id in backend
```
Comment (Murray-LIANG) [inline service/controller.go]: in-consistent with the implementation?

File: service/controller.go [hunk 2]
```diff
@@ -14,6 +19,65 @@ const (
...
+	}
+
+	if !completed {
+		log.Error("Not completed in time.")
+		jobErr = errors.New(fmt.Sprintf("Unity job %s not completed in time", jobId))
+	}
+	return completed, state, jobErr
+}
+
 //TODO: not enough for hack attack
```
Comment (Murray-LIANG) [inline service/controller.go]: Help remove the `TODO` comment.

File: service/controller.go [hunk 3]
```diff
@@ -28,6 +92,9 @@ func (s *service) CreateVolume(
 		fsType := mountCap.FsType
 		log.Info("FsType:", fsType)
 		fileResp, fileErr := FileCreateVolume(s, ctx, req)
+		if fileErr == nil {
```
Comment (Murray-LIANG) [inline service/controller.go:94]: Do we have any `volume-existing` error? We should only catch the `volume-existing` error.

File: service/controller.go [hunk 4]
```diff
@@ -63,23 +130,30 @@ func (s *service) DeleteVolume(
...
 	}
+	backendId, volType := getBackendIdAndTypeByVolumeId(volId)
+
+	deleteResponse := &csi.DeleteVolumeResponse{}
+	var err error = nil
+	if volType == Volume_Type_File {
+		err = FileDeleteVolume(s, ctx, backendId)
+	}
 
 	// TODO (ryan) add the implementation
```
Comment (Murray-LIANG) [inline service/controller.go]: Remove this comment.

File: service/controller.go (hunk 3 above)
Comment (qingfeng0820) [reply service/controller.go:94]: I didn't touch the FileCreateVolume function, I just add the file volume prefix when the function returned successfully. I can look at that function next.

File: service/controller.go (hunk 1 above)
Comment (qingfeng0820) [reply service/controller.go]: Where is in-consistent? This method is used to remove the prefix of volume id .
If it's block type, this method returns block_id (lun_id).
If it's file type, this method returns file_id:share_id.

**Pull request merged**